docs/proposals/local-cluster-ux.md

```diff
@@ -0,0 +1,184 @@
+This proposal is a RFC!
+
+## Primary Goals
+
+From a high level the goal is to make it easy for a new user to run a Kubernetes cluster and play with curated examples that require least amount of knowledge about Kubernetes.
+These examples will only use kubectl and only a subset of Kubernetes features that are available will be exposed.
+
+- Works across multiple OSes - OS X, Linux and Windows primarily.
+- Single command setup and teardown UX.
+- Unified UX across OSes
+- Minimal dependencies on third party software.
+- Minimal resource overhead.
+- Eliminate any other alternatives to local cluster deployment.
+
+## Secondary Goals
+
+- Enable developers to use the local cluster for kubernetes development.
+
+## Non Goals
+
+- Simplifying kubernetes production deployment experience
+- Supporting all possible deployment configurations of Kubernetes like various types of storage, networking, etc.
+
+
+## Local cluster requirements
+
+- Includes all the master components & DNS (Apiserver, scheduler, controller manager, etcd and kube dns)
+- Basic auth
+- Service accounts should be setup
+- Kubectl should be auto-configured to use the local cluster
+- Tested & maintained as part of Kubernetes core
+
+## Existing solutions
+
+Following are some of the existing solutions that attempt to simplify local cluster deployments.
+
+### [Spread](https://github.com/redspread/spread)
+
+Spread's UX is great!
+It is adapted from monokube and includes DNS as well.
+It satisfies almost all the requirements, excepting that of requiring docker to be pre-installed.
+It has a loose dependency on docker.
+New releases of docker might break this setup.
+
+### [Kmachine](https://github.com/skippbox/kmachine)
+
+Kmachine is adapted from docker-machine.
+It exposes the entire docker-machine CLI.
+It is possible to repurpose Kmachine to meet all our requirements.
+
+### [Monokube](https://github.com/polvi/monokube)
+
+Single binary that runs all kube master components.
+Does not include DNS.
+This is only a part of the overall local cluster solution.
+
+### Vagrant
+
+The kube-up.sh script included in Kubernetes release supports a few Vagrant based local cluster deployments.
+kube-up.sh is not user friendly.
+It typically takes a long time for the cluster to be set up using vagrant and often times is unsuccessful on OS X.
+The [Core OS single machine guide](https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant-single.html)  uses Vagrant as well and it just works.
+Since we are targeting a single command install/teardown experience, vagrant needs to be an implementation detail and not be exposed to our users.
+
+## Proposed Solution
+
+To avoid exposing users to third party software and external dependencies, we will build a toolbox that will be shipped with all the dependencies including all kubernetes components, hypervisor, base image, kubectl, etc.
+*Note: Docker provides a [similar toolbox](https://www.docker.com/products/docker-toolbox).*
```

Comment:
I don't want to sound self-serving here but definitely a kubernetes toolbox is needed.

While there are ways to run k8s with a single binary now, as well as run Docker on OSX natively, I think that a toolbox can provide a single user experience by packaging whatever is needed.

So +1

At skippbox we started with this idea and got inspired by the docker toolbox, so we created

* [boot2k8s](https://github.com/skippbox/boot2k8s)
* [kmachine](https://github.com/skippbox/kmachine)
* [kompose](https://github.com/skippbox/kompose) 
* [skippbox](https://github.com/skippbox/skippbox) a kitematic for k8s

We had an early prototype of packaging everything in a [toolbox](https://github.com/skippbox/toolbox) including kubectl.